File: goclient/item/services_json_request_builder.go
package item

import (
    "context"
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
)

// ServicesJsonRequestBuilder builds and executes requests for operations under \{provider-id}\services.json
type ServicesJsonRequestBuilder struct {
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.BaseRequestBuilder
}
// ServicesJsonRequestBuilderGetRequestConfiguration configuration for the request such as headers, query parameters, and middleware options.
type ServicesJsonRequestBuilderGetRequestConfiguration struct {
    // Request headers
    Headers *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestHeaders
    // Request options
    Options []i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestOption
}
// NewServicesJsonRequestBuilderInternal instantiates a new ServicesJsonRequestBuilder and sets the default values.
func NewServicesJsonRequestBuilderInternal(pathParameters map[string]string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*ServicesJsonRequestBuilder) {
    m := &ServicesJsonRequestBuilder{
        BaseRequestBuilder: *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewBaseRequestBuilder(requestAdapter, "{+baseurl}/{provider%2Did}/services.json", pathParameters),
    }
    return m
}
// NewServicesJsonRequestBuilder instantiates a new ServicesJsonRequestBuilder and sets the default values.
func NewServicesJsonRequestBuilder(rawUrl string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*ServicesJsonRequestBuilder) {
    urlParams := make(map[string]string)
    urlParams["request-raw-url"] = rawUrl
    return NewServicesJsonRequestBuilderInternal(urlParams, requestAdapter)
}
// Get list all serviceNames in the directory for a particular providerName
// Deprecated: This method is obsolete. Use GetAsServicesGetResponse instead.
// returns a ServicesJsonServicesResponseable when successful
func (m *ServicesJsonRequestBuilder) Get(ctx context.Context, requestConfiguration *ServicesJsonRequestBuilderGetRequestConfiguration)(ServicesJsonServicesResponseable, error) {
    requestInfo, err := m.ToGetRequestInformation(ctx, requestConfiguration);
    if err != nil {
        return nil, err
    }
    res, err := m.BaseRequestBuilder.RequestAdapter.Send(ctx, requestInfo, CreateServicesJsonServicesResponseFromDiscriminatorValue, nil)
    if err != nil {
        return nil, err
    }
    if res == nil {
        return nil, nil
    }
    return res.(ServicesJsonServicesResponseable), nil
}
// GetAsServicesGetResponse list all serviceNames in the directory for a particular providerName
// returns a ServicesJsonServicesGetResponseable when successful
func (m *ServicesJsonRequestBuilder) GetAsServicesGetResponse(ctx context.Context, requestConfiguration *ServicesJsonRequestBuilderGetRequestConfiguration)(ServicesJsonServicesGetResponseable, error) {
    requestInfo, err := m.ToGetRequestInformation(ctx, requestConfiguration);
    if err != nil {
        return nil, err
    }
    res, err := m.BaseRequestBuilder.RequestAdapter.Send(ctx, requestInfo, CreateServicesJsonServicesGetResponseFromDiscriminatorValue, nil)
    if err != nil {
        return nil, err
    }
    if res == nil {
        return nil, nil
    }
    return res.(ServicesJsonServicesGetResponseable), nil
}
// ToGetRequestInformation list all serviceNames in the directory for a particular providerName
// returns a *RequestInformation when successful
func (m *ServicesJsonRequestBuilder) ToGetRequestInformation(ctx context.Context, requestConfiguration *ServicesJsonRequestBuilderGetRequestConfiguration)(*i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestInformation, error) {
    requestInfo := i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewRequestInformationWithMethodAndUrlTemplateAndPathParameters(i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.GET, m.BaseRequestBuilder.UrlTemplate, m.BaseRequestBuilder.PathParameters)
    if requestConfiguration != nil {
        requestInfo.Headers.AddAll(requestConfiguration.Headers)
        requestInfo.AddRequestOptions(requestConfiguration.Options)
    }
    requestInfo.Headers.TryAdd("Accept", "application/json")
    return requestInfo, nil
}
// WithUrl returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
// returns a *ServicesJsonRequestBuilder when successful
func (m *ServicesJsonRequestBuilder) WithUrl(rawUrl string)(*ServicesJsonRequestBuilder) {
    return NewServicesJsonRequestBuilder(rawUrl, m.BaseRequestBuilder.RequestAdapter);
}

File: javaclient/app/src/main/java/javaclient/app/item/servicesjson/ServicesJsonRequestBuilder.java
package javaclient.client.item.servicesjson;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.BaseRequestConfiguration;
import com.microsoft.kiota.HttpMethod;
import com.microsoft.kiota.RequestAdapter;
import com.microsoft.kiota.RequestInformation;
import com.microsoft.kiota.RequestOption;
import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParsableFactory;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /{provider-id}/services.json
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class ServicesJsonRequestBuilder extends BaseRequestBuilder {
    /**
     * Instantiates a new {@link ServicesJsonRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ServicesJsonRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/{provider%2Did}/services.json", pathParameters);
    }
    /**
     * Instantiates a new {@link ServicesJsonRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ServicesJsonRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/{provider%2Did}/services.json", rawUrl);
    }
    /**
     * List all serviceNames in the directory for a particular providerName
     * @return a {@link ServicesGetResponse}
     */
    @jakarta.annotation.Nullable
    public ServicesGetResponse get() {
        return get(null);
    }
    /**
     * List all serviceNames in the directory for a particular providerName
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link ServicesGetResponse}
     */
    @jakarta.annotation.Nullable
    public ServicesGetResponse get(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = toGetRequestInformation(requestConfiguration);
        return this.requestAdapter.send(requestInfo, null, ServicesGetResponse::createFromDiscriminatorValue);
    }
    /**
     * List all serviceNames in the directory for a particular providerName
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation() {
        return toGetRequestInformation(null);
    }
    /**
     * List all serviceNames in the directory for a particular providerName
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = new RequestInformation(HttpMethod.GET, urlTemplate, pathParameters);
        requestInfo.configure(requestConfiguration, GetRequestConfiguration::new);
        requestInfo.headers.tryAdd("Accept", "application/json");
        return requestInfo;
    }
    /**
     * Returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
     * @param rawUrl The raw URL to use for the request builder.
     * @return a {@link ServicesJsonRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ServicesJsonRequestBuilder withUrl(@jakarta.annotation.Nonnull final String rawUrl) {
        Objects.requireNonNull(rawUrl);
        return new ServicesJsonRequestBuilder(rawUrl, requestAdapter);
    }
    /**
     * Configuration for the request such as headers, query parameters, and middleware options.
     */
    @jakarta.annotation.Generated("com.microsoft.kiota")
    public class GetRequestConfiguration extends BaseRequestConfiguration {
    }
}
